[ER] Report crash while checkout resource deleted

Hi team,

I debug the provider in cn-north-7 and the provider crashed.
The reason is the filter result reference is incorrect.

The error message for provider crash log:
```
Stack trace from the terraform-provider-huaweicloud_v1.49.0 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 454 [running]:
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/er.QueryPropagationById(0xc0005a9400?, {0xc000d488d0?, 0x2?}, ...
    github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/er/resource_huaweicloud_er_propagation.go:148
```

best regards!

--- a/huaweicloud/services/er/resource_huaweicloud_er_association.go
+++ b/huaweicloud/services/er/resource_huaweicloud_er_association.go
@@ -137,7 +137,7 @@ func QueryAssociationById(client *golangsdk.ServiceClient, instanceId, routeTabl
 	if err != nil {
 		return nil, err
 	}
-	if len(resp) < 1 {
+	if len(result) < 1 {
 		return nil, golangsdk.ErrDefault404{
 			ErrUnexpectedResponseCode: golangsdk.ErrUnexpectedResponseCode{
 				Body: []byte(fmt.Sprintf("the association (%s) does not exist", associationId)),
@@ -145,8 +145,12 @@ func QueryAssociationById(client *golangsdk.ServiceClient, instanceId, routeTabl
 		}
 	}
 
-	association := result[0].(associations.Association)
-	log.Printf("[DEBUG] The details of the association (%s) is: %#v", associationId, association)
+	log.Printf("[DEBUG] The result filtered by resource ID (%s) is: %#v", associationId, result)
+	association, ok := result[0].(associations.Association)
+	if !ok {
+		return nil, fmt.Errorf("the element type of filter result is incorrect, want 'associations.Association', "+
+			"but got '%T'", result[0])
+	}
 
 	return &association, nil
 }

--- a/huaweicloud/services/er/resource_huaweicloud_er_propagation.go
+++ b/huaweicloud/services/er/resource_huaweicloud_er_propagation.go
@@ -145,8 +145,12 @@ func QueryPropagationById(client *golangsdk.ServiceClient, instanceId, routeTabl
 		}
 	}
 
-	association := result[0].(propagations.Propagation)
-	log.Printf("[DEBUG] The details of the propagation (%s) is: %#v", propagationId, association)
+	log.Printf("[DEBUG] The result filtered by resource ID (%s) is: %#v", propagationId, result)
+	association, ok := result[0].(propagations.Propagation)
+	if !ok {
+		return nil, fmt.Errorf("the element type of filter result is incorrect, want 'propagations.Propagation', "+
+			"but got '%T'", result[0])
+	}
 
 	return &association, nil
 }

